server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -2604,4 +2614,12 @@ public void test_match_with_geo_shape_is_streamed_as_text_type_to_4_1_8_nodes()
         Function serializedTo41 = new Function(in);
         assertThat(serializedTo41.info().ident().argumentTypes().get(1), is(DataTypes.STRING));
     }
+
+    @Test
+    public void test_table_function_wrapped_inside_scalar_can_be_used_inside_group_by() {
+        var executor = SQLExecutor.builder(clusterService)
+            .build();
+        AnalyzedRelation rel = executor.analyze("select regexp_matches('foo', '.*')[1] from sys.cluster group by 1");
```

Comment:
Should we also add a logical planner test for this case? Otherwise the new ProjectSet creation change is not tested, or am I missing something?